### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by hdonnay**
This fixes the comments and uses the url package to resolve references.

### CODE REVIEW

**Comment by ldelossa** [comment]
@hdonnay can license be moved to a single file at root of repo?

**Comment by hdonnay** [comment]
I dunno, the [apache site](https://www.apache.org/licenses/LICENSE-2.0) says to use a header.

**Review by ldelossa — APPROVED** [review]

File: repomd.go [hunk 1]
```diff
@@ -62,14 +63,21 @@ type Location struct {
 // If a mirror url is provided a fully qualified Repo.Location.Href is returned
 // A ErrRepoNotFound error is returned if the RepoType cannot be located.
 func (md *RepoMD) Repo(t RepoType, mirror string) (Repo, error) {
-	var repo Repo
 	for _, repo := range md.RepoList {
 		if repo.Type == string(t) {
 			if mirror != "" {
-				repo.Location.Href = fmt.Sprintf("%s%s", mirror, repo.Location.Href)
+				u, err := url.Parse(mirror)
```
Comment (ldelossa) [inline repomd.go:69]: :+1: :+1:

**Review by ldelossa — APPROVED** [review]

**Comment by ldelossa** [comment]
@hdonnay I'll bump the release when you merge

**Pull request merged**